boto3/crt.py

```diff
@@ -157,11 +173,45 @@ def compare_identity(boto3_creds, crt_s3_creds):
     return is_matching_identity
 
 
+def _validate_crt_transfer_config(config):
+    if config is None:
+        return
+    # CRT client can also be configured via `AUTO_RESOLVE_TRANSFER_CLIENT`
+    # but it predates this validation. We only validate against CRT client
+    # configured via `CRT_TRANSFER_CLIENT` to preserve compatibility.
+    if config.preferred_transfer_client != CRT_TRANSFER_CLIENT:
+        return
+    invalid_crt_args = []
+    for param in config.DEFAULTS.keys():
+        val = config.get_deep_attr(param)
+        if (
+            param not in _ALLOWED_CRT_TRANSFER_CONFIG_OPTIONS
+            and val is not config.UNSET_DEFAULT
+        ):
+            invalid_crt_args.append(param)
+    if len(invalid_crt_args) > 0:
+        raise InvalidCrtTransferConfigError(
+            "The following transfer config options are invalid "
+            "when `preferred_transfer_client` is set to `crt`: `"
+            f"{'`, `'.join(invalid_crt_args)}`"
```

Comment:
nit; I think this is our only error using ` for formatting. Markdown won't render in the terminal/logs, so just want to make sure this was intentional.